platforms/documentation/docs/src/docs/userguide/dep-man/04-modeling-features/artifact_transforms.adoc

```diff
@@ -228,39 +403,44 @@ There are a few points to consider when using `registerTransform()`:
 For example, imagine you want to unpack some dependencies and put the unpacked directories and files on the classpath.
 You can do so by registering an artifact transform action of type `Unzip`, as shown here:
 
-.Artifact transform registration without parameters
 ====
 include::sample[dir="snippets/dependencyManagement/artifactTransforms-unzip/kotlin",files="build.gradle.kts[tags=artifact-transform-registration]"]
 include::sample[dir="snippets/dependencyManagement/artifactTransforms-unzip/groovy",files="build.gradle[tags=artifact-transform-registration]"]
 ====
 
 Another example is that you want to minify JARs by only keeping some `class` files from them.
-Note the use of the `parameters {}` block to provide the classes to keep in the minified JARs to the `Minify` transform.
+Note the use of the `parameters {}` block to provide the classes to keep in the minified JARs to the `Minify` transform:
 
-.Artifact transform registration with parameters
 ====
 include::sample[dir="snippets/dependencyManagement/artifactTransforms-minify/kotlin",files="build.gradle.kts[tags=artifact-transform-registration]"]
 include::sample[dir="snippets/dependencyManagement/artifactTransforms-minify/groovy",files="build.gradle[tags=artifact-transform-registration]"]
 ====
 
-== Implementing incremental artifact transforms
+== Executing Artifact Transforms
 
-Similar to <<custom_tasks#incremental_tasks,incremental tasks>>, artifact transforms can avoid work by only processing changed files from the last execution.
-This is done by using the link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] interface.
-For artifact transforms, only the input artifact is an incremental input, and therefore the transform can only query for changes there.
-In order to use link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] in the transform action, inject it into the action.
-For more information on how to use link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges], see the corresponding documentation for <<custom_tasks#incremental_tasks,incremental tasks>>.
+On the command line, Gradle runs tasks; not Artifact Transforms: `./gradlew build.`
+So how and when does it run transforms?
 
-Here is an example of an incremental transform that counts the lines of code in Java source files:
+A task can declare an input with attributes.
+When running this task, Gradle may run an Artifact Transform to ensure that the input has the proper attributes.
+Then, it will run the task.
```

Comment:
This whole paragraph seems a bit imprecise to me. What do you mean by `A task can declare an input with attributes.`? Tasks can declare `FileCollections` as inputs, which are derived from resolvable configurations. Those configurations have attributes and may run artifact transforms during resolution.

Something about "input with attributes" does not sound right to me.